src/arfs_file_wrapper.ts

```diff
@@ -127,6 +130,70 @@ export class ArFSFolderToUpload {
 		}
 	}
 
+	public async checkAndAssignExistingNames(
+		getExistingNamesFn: (parentFolderId: FolderID) => Promise<EntityNamesAndIds>
+	): Promise<void> {
+		if (!this.existingId) {
+			// Folder has no existing ID to check
+			return;
+		}
+
+		const existingEntityNamesAndIds = await getExistingNamesFn(this.existingId);
+
+		for await (const file of this.files) {
+			const baseFileName = file.getBaseFileName();
+
+			const folderNameConflict = existingEntityNamesAndIds.folders.find(
```

Comment:
One note on the name choice: It's not clear from the variable name whether `folder` in `folderNameConflict` is the source folder or the destination folder. Consider `existingFolderAtDestConflict`. Similar concept can be applied to all other conflicts.